### PR TITLE
Track manually completed sub-measurements

### PR DESCRIPTION
**Pull request by konstantin-msft**
- Track manually completed sub-measurements.
- Add unit tests.

### CODE REVIEW

File: lib/msal-common/src/telemetry/performance/PerformanceClient.ts [hunk 1]
```diff
@@ -318,6 +318,7 @@ export abstract class PerformanceClient implements IPerformanceClient {
 
                     const completedEvent = this.endMeasurement(event);
                     if (completedEvent) {
+                        completedEvent.isIncomplete = true;
```
Comment (peterzenz) [inline lib/msal-common/src/telemetry/performance/PerformanceClient.ts]: I'd expect that an incomplete measurement represents either a bug, a timeout or crash somewhere.  Please add a datapoint to log whether we had incomplete measurements on a given call.  Just one datapoint for now, if we find there are tons of them we'll look at something more comprehensive if needed so we can debug/understand.

**Review by peterzenz — COMMENTED** [review]
I've asked for an additional datapoint. You can complete this without it, but please follow up with another change if you choose to go that route.

File: lib/msal-common/src/telemetry/performance/PerformanceClient.ts (hunk 1 above)
Comment (konstantin-msft) [reply lib/msal-common/src/telemetry/performance/PerformanceClient.ts]: Would a single data point with an array of manually completed events work? This should be both inexpensive to store and easy to debug.

Comment (peterzenz) [reply lib/msal-common/src/telemetry/performance/PerformanceClient.ts]: Very easy to debug, expensive to store in telemetry.  For telemetry a boolean just to see how prevalent it is, would be great. 
For tracing/logging, an array of manually completed events would be fine and that would help with one-off debugging in the near term.

Comment (konstantin-msft) [reply lib/msal-common/src/telemetry/performance/PerformanceClient.ts]: Replaced the previous approach with a counter for incomplete sub-measurements.

File: lib/msal-common/test/telemetry/PerformanceClient.spec.ts [hunk 2]
```diff
@@ -3,9 +3,16 @@
  * Licensed under the MIT License.
  */
 
-import { ApplicationTelemetry, IGuidGenerator, IPerformanceMeasurement, Logger, PerformanceEvents } from "../../src";
-import { IPerformanceClient } from "../../src/telemetry/performance/IPerformanceClient";
-import { PerformanceClient } from "../../src/telemetry/performance/PerformanceClient";
+import {
```
Comment (tnorling) [inline lib/msal-common/test/telemetry/PerformanceClient.spec.ts:6]: Import from the relevant file(s), not src

**Review by tnorling — APPROVED** [review]

File: lib/msal-common/test/telemetry/PerformanceClient.spec.ts (hunk 2 above)
Comment (konstantin-msft) [reply lib/msal-common/test/telemetry/PerformanceClient.spec.ts:6]: Good catch, thank you Thomas. Fixed.

**Review by peterzenz — APPROVED** [review]

**Pull request merged**